File: app/src/main/java/com/pandacorp/taskui/ui/settings/SettingsActivity.java
package com.pandacorp.taskui.ui.settings;

import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.preference.ListPreference;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceManager;

import com.pandacorp.taskui.MainActivity;
import com.pandacorp.taskui.R;
import com.pandacorp.taskui.Widget.WidgetProvider;

public class SettingsActivity extends AppCompatActivity {
    final String TAG = "MyLogs";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        new MySettings(this).load();
        setContentView(R.layout.fragment_settings);
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.frameLayout, new SettingsFragment())
                .commit();
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeButtonEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home: //Метод обработки нажатия на кнопку home.
                startActivity(new Intent(this, MainActivity.class));
                finish();

                return super.onOptionsItemSelected(item);
            default:
                return super.onOptionsItemSelected(item);
        }

    }

    public static class SettingsFragment extends PreferenceFragmentCompat implements Preference.OnPreferenceChangeListener {

        SharedPreferences sp;

        ListPreference themes_listPreference;
        ListPreference languages_listPreference;
        Preference version_Preference;

        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            setPreferencesFromResource(R.xml.root_preferences, rootKey);

            sp = PreferenceManager.getDefaultSharedPreferences(getContext());

            themes_listPreference = findPreference("Themes");
            themes_listPreference.setOnPreferenceChangeListener(this);

            languages_listPreference = findPreference("Languages");
            languages_listPreference.setOnPreferenceChangeListener(this);

            version_Preference = findPreference("Version");

            String version_name;
            try {
                version_name = getContext().getPackageManager().getPackageInfo(getContext().getPackageName(), 0).versionName;
                version_Preference.setTitle(getResources().getString(R.string.version) + " " + version_name);
                //Тут происходит добавление загаловка в виде версии к пункту настроек.

            } catch (PackageManager.NameNotFoundException e) {
                e.printStackTrace();
            }
        }

        @Override
        public boolean onPreferenceChange(Preference preference, Object newValue) {

            WidgetProvider.Companion.sendRefreshBroadcast(requireContext());
            getActivity().recreate();

            return true;
        }



    }



}

File: app/src/main/java/com/pandacorp/taskui/presentation/ui/settings/SettingsActivity.kt
package com.pandacorp.taskui.presentation.ui.settings

import android.content.SharedPreferences
import android.content.pm.PackageManager
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.ListPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.PreferenceManager
import com.pandacorp.taskui.R
import com.pandacorp.taskui.databinding.ActivitySettingsBinding
import com.pandacorp.taskui.presentation.ui.settings.dialogs.DialogListView
import com.pandacorp.taskui.presentation.utils.Constants
import com.pandacorp.taskui.presentation.utils.PreferenceHandler
import com.pandacorp.taskui.presentation.utils.getPackageInfoCompat

class SettingsActivity : AppCompatActivity() {
    private lateinit var sp: SharedPreferences

    private var _binding: ActivitySettingsBinding? = null
    private val binding get() = _binding!!

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        PreferenceHandler(this).load()
        _binding = ActivitySettingsBinding.inflate(layoutInflater)
        setContentView(binding.root)
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.settings, SettingsFragment())
            .commit()
        setSupportActionBar(findViewById(R.id.toolbar))
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeButtonEnabled(true)
        supportActionBar?.setTitle(R.string.settings)
        sp = PreferenceManager.getDefaultSharedPreferences(this)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
            }
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onDestroy() {
        _binding = null
        super.onDestroy()
    }

    class SettingsFragment : PreferenceFragmentCompat() {
        private lateinit var sp: SharedPreferences
        private var themesListPreference: ListPreference? = null
        private var languagesListPreference: ListPreference? = null
        private var versionPreference: Preference? = null
        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.root_preferences, rootKey)
            sp = PreferenceManager.getDefaultSharedPreferences(requireContext())
            themesListPreference = findPreference(Constants.PreferencesKeys.themesKey)
            languagesListPreference = findPreference(Constants.PreferencesKeys.languagesKey)
            versionPreference = findPreference(Constants.PreferencesKeys.versionKey)
            // Here we add version title to this preference, we get version from build.gradle file.
            try {
                versionPreference!!.title =
                    resources.getString(R.string.version) + " " + requireContext().packageManager
                        .getPackageInfoCompat(requireContext().packageName).versionName
            } catch (e: PackageManager.NameNotFoundException) {
                e.printStackTrace()
            }
        }

        override fun onDisplayPreferenceDialog(preference: Preference) {
            when (preference.key) {
                Constants.PreferencesKeys.themesKey -> // rounded themes dialog with images
                    DialogListView.newInstance(Constants.PreferencesKeys.themesKey)
                        .show(parentFragmentManager, null)

                Constants.PreferencesKeys.languagesKey -> // rounded languages dialog with images
                    DialogListView.newInstance(Constants.PreferencesKeys.languagesKey)
                        .show(parentFragmentManager, null)

                else -> {
                    super.onDisplayPreferenceDialog(preference)
                }
            }

        }

    }

    companion object {
        const val TAG = "SettingsActivity"
    }
}